packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -0,0 +1,146 @@
+                        .getConfiguration()
+                        .get('aws.accessAnalyzer.policyChecks.cloudFormationParameterFilePath')!
+                )
+            }
+        })
+    }
+
+    // TODO: Add functionality for the three checks
+    public async validatePolicy(documentType: string, policyType: string) {
+        this.client /* Do something */
+    }
+    public async checkNoNewAccess() {}
+    public async checkAccessNotGranted() {}
+}
+
+const Panel = VueWebview.compilePanel(IamPolicyChecksWebview)
+
+export async function renderIamPolicyChecks(context: ExtContext): Promise<void> {
+    const logger: Logger = getLogger()
+    try {
+        const client = new AccessAnalyzer({ region: context.regionProvider.defaultRegionId })
+        //Read from settings to auto-fill some inputs
+        const referencePolicyFilePath: string = vscode.workspace
+            .getConfiguration()
+            .get('aws.accessAnalyzer.policyChecks.referencePolicyFilePath')!
+        const cfnParameterPath: string = vscode.workspace
+            .getConfiguration()
+            .get('aws.accessAnalyzer.policyChecks.cloudFormationParameterFilePath')!
+
+        //Check if Cfn and Tf tools are installed
+        var cfnToolInstalled = true
+        var tfToolInstalled = true
+        exec('tf-policy-validator', err => {
+            if (err) {
+                if (err.message.includes('command not found')) {
+                    tfToolInstalled = false
+                }
+            }
+        })
+        exec('cfn-policy-validator', err => {
+            if (err) {
+                if (err.message.includes('command not found')) {
+                    cfnToolInstalled = false
+                }
+            }
+        })
+
+        const wv = new Panel(
+            context.extensionContext,
+            {
+                referenceFilePath: referencePolicyFilePath ? referencePolicyFilePath : '',
+                cfnParameterPath: cfnParameterPath ? cfnParameterPath : '',
+                referenceDocument: await _getReferenceDocument(referencePolicyFilePath),
+                pythonToolsInstalled: cfnToolInstalled && tfToolInstalled,
+            },
+            client
+        )
+        await wv.show({
+            viewColumn: vscode.ViewColumn.Beside,
+            title: localize('AWS.iamPolicyChecks.title', 'IAM Policy Checks'),
+        })
+    } catch (err) {
+        logger.error(err as Error)
+    }
+}
+
+// Helper function to get document contents from a path
+async function _getReferenceDocument(input: string): Promise<string> {
+    if (fs.existsSync(input)) {
+        return fs.readFileSync(input).toString()
+    } else {
+        try {
+            const { region, bucket, key } = AmazonS3URI(input)
+            const s3Client = new S3({ region })
+            const resp = await s3Client.getObject({ Bucket: bucket, Key: key }).promise()
+            return resp.Body!.toString()
+        } catch (e: any) {}
+    }
+    return ''
```

Comment:
Q: is this not considered to be an error?

Reply:
It is, and in this version I decided not to surface these errors because I was afraid it was too noisy if a user were to change their reference path frequently. I will be updating this to surface errors in a more user-friendly way.

Reply:
`because I was afraid it was too noisy if a user were to change their reference path frequently.`, if the changes cannot be resolved to a correct reference document, the plugin would simply not work, and user should be aware that this is an error right?

Or maybe you can elaborate a bit on what's your thought on `surface errors in a more user-friendly way.` mean?

Reply:
I was planning to surface all errors with notifications, which would be pretty annoying if you're typing out the path and notifications keep popping up. In the above, we display a "preview" of the file; if it is empty then I assumed users would be able to tell that it is not being read.

I am going to surface errors in the Webview itself, as an error message next to the input path.